src/main/java/com/amazon/opendistroforelasticsearch/security/filter/OpenDistroSecurityFilter.java

```diff
@@ -330,12 +335,27 @@ private boolean checkImmutableIndices(Object request, ActionListener listener) {
         }
         
         if(request instanceof IndexRequest) {
-            if(complianceConfig != null && complianceConfig.isIndexImmutable(request)) {
+            if (isIndexImmutable(request)) {
                 ((IndexRequest) request).opType(OpType.CREATE);
             }
         }
         
         return false;
     }
 
-}
\ No newline at end of file
+    private boolean isIndexImmutable(Object request) {
```

Comment:
nit: consider `isRequestIndexImmutable`

Reply:
Retained the existing function name. Wanted to reduce diffs

Reply:
This is a new function, the diff is already there.